### ISSUE TITLE
enum value defines included from a separate header don't show up in output

### ISSUE THREAD

**Issue by oumuamua97** [doxygen/doxygen#10272]
**Describe the bug**
Using the inclusion pattern below to define enum values, the enum values don't show up in the xml output:

```
// specialcode.h:

#define SPECIAL_CODE(name, value, doc) name = (value),

enum special_codes
{
#include "specialcodes.h"
};
```

```
// specialcodes.h:

SPECIAL_CODE(SC_ORDINARY, 0, "not unlike others")
SPECIAL_CODE(SC_SPECIAL, 1, "something different")
SPECIAL_CODE(SC_UNKNOWN, 2, "completely unheard of")
```

This used to work as expected up until the following commit:

commit 047df30104f628bdfbbe71fbc8d11440a5b0494a
Date:   Sun Jun 2 13:46:31 2019 +0200

    issue #7021 Warning about missing item in INCLUDE_PATH

    Do the `curlyCount` per file and not globally. 

This is the resulting xml:

```
<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.8.16">
  <compounddef id="specialcode_8h" kind="file" language="C++">
    <compoundname>specialcode.h</compoundname>
    <includes local="yes">specialcodes.h</includes>
    <incdepgraph>
      <node id="1">
        <label>specialcode.h</label>
        <link refid="specialcode_8h"/>
        <childnode refid="2" relation="include">
        </childnode>
      </node>
      <node id="2">
        <label>specialcodes.h</label>
      </node>
    </incdepgraph>
      <sectiondef kind="define">
      <memberdef kind="define" id="specialcode_8h_1a316aaab1be3b34f9d97409001d8ea150" prot="public" static="no">
        <name>SPECIAL_CODE</name>
        <param><defname>name</defname></param>
        <param><defname>value</defname></param>
        <param><defname>doc</defname></param>
        <initializer>name = (value),</initializer>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="specialcode.h" line="4" column="9" bodyfile="specialcode.h" bodystart="4" bodyend="-1"/>
      </memberdef>
      </sectiondef>
      <sectiondef kind="enum">
      <memberdef kind="enum" id="specialcode_8h_1a7a0301ebc6de6bd679e0660bbf29416d" prot="public" static="no" strong="no">
        <type></type>
        <name>special_codes</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="specialcode.h" line="7" column="1" bodyfile="specialcode.h" bodystart="6" bodyend="9"/>
      </memberdef>
      </sectiondef>
    <briefdescription>
    </briefdescription>
    <detaileddescription>
    </detaileddescription>
    <programlisting>
<codeline lineno="1"><highlight class="preprocessor">#ifndef<sp/>_SPECIAL_CODE_H</highlight><highlight class="normal"></highlight></codeline>
<codeline lineno="2"><highlight class="normal"></highlight><highlight class="preprocessor">#define<sp/>_SPECIAL_CODE_H</highlight><highlight class="normal"></highlight></codeline>
<codeline lineno="3"><highlight class="normal"></highlight></codeline>
<codeline lineno="4"><highlight class="normal"></highlight><highlight class="preprocessor">#define<sp/>SPECIAL_CODE(name,<sp/>value,<sp/>doc)<sp/>name<sp/>=<sp/>(value),</highlight><highlight class="normal"></highlight></codeline>
<codeline lineno="5"><highlight class="normal"></highlight></codeline>
<codeline lineno="6"><highlight class="normal"></highlight><highlight class="keyword">enum</highlight><highlight class="normal"><sp/>special_codes</highlight></codeline>
<codeline lineno="7"><highlight class="normal">{</highlight></codeline>
<codeline lineno="8"><highlight class="normal"></highlight><highlight class="preprocessor">#include<sp/>&quot;specialcodes.h&quot;</highlight><highlight class="normal"></highlight></codeline>
<codeline lineno="9"><highlight class="normal">};</highlight></codeline>
<codeline lineno="10"><highlight class="normal"></highlight></codeline>
<codeline lineno="11"><highlight class="normal"></highlight><highlight class="preprocessor">#endif</highlight></codeline>
    </programlisting>
    <location file="specialcode.h"/>
  </compounddef>
</doxygen>
```

**Expected behavior**
Here's the xml output with the preceding commit (7403c9e4f), in which the enum values do show up as expected:
```
<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.8.16">
  <compounddef id="specialcode_8h" kind="file" language="C++">
    <compoundname>specialcode.h</compoundname>
    <includes local="yes">specialcodes.h</includes>
    <incdepgraph>
      <node id="1">
        <label>specialcode.h</label>
        <link refid="specialcode_8h"/>
        <childnode refid="2" relation="include">
        </childnode>
      </node>
      <node id="2">
        <label>specialcodes.h</label>
      </node>
    </incdepgraph>
      <sectiondef kind="define">
      <memberdef kind="define" id="specialcode_8h_1a316aaab1be3b34f9d97409001d8ea150" prot="public" static="no">
        <name>SPECIAL_CODE</name>
        <param><defname>name</defname></param>
        <param><defname>value</defname></param>
        <param><defname>doc</defname></param>
        <initializer>name = (value),</initializer>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="specialcode.h" line="4" column="9" bodyfile="specialcode.h" bodystart="4" bodyend="-1"/>
      </memberdef>
      </sectiondef>
      <sectiondef kind="enum">
      <memberdef kind="enum" id="specialcode_8h_1a7a0301ebc6de6bd679e0660bbf29416d" prot="public" static="no" strong="no">
        <type></type>
        <name>special_codes</name>
        <enumvalue id="specialcode_8h_1a7a0301ebc6de6bd679e0660bbf29416da5ed60f4cb5eac680e66122273f9e31cb" prot="public">
          <name>SC_ORDINARY</name>
          <initializer>= ( 0 )</initializer>
          <briefdescription>
          </briefdescription>
          <detaileddescription>
          </detaileddescription>
        </enumvalue>
        <enumvalue id="specialcode_8h_1a7a0301ebc6de6bd679e0660bbf29416da81412a84e50df7803d94a25b2788dc38" prot="public">
          <name>SC_SPECIAL</name>
          <initializer>= ( 1 )</initializer>
          <briefdescription>
          </briefdescription>
          <detaileddescription>
          </detaileddescription>
        </enumvalue>
        <enumvalue id="specialcode_8h_1a7a0301ebc6de6bd679e0660bbf29416da2f0d9d921e4a4df960785588bff35b69" prot="public">
          <name>SC_UNKNOWN</name>
          <initializer>= ( 2 )</initializer>
          <briefdescription>
          </briefdescription>
          <detaileddescription>
          </detaileddescription>
        </enumvalue>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="specialcode.h" line="7" column="1" bodyfile="specialcode.h" bodystart="6" bodyend="9"/>
      </memberdef>
      </sectiondef>
    <briefdescription>
    </briefdescription>
    <detaileddescription>
    </detaileddescription>
    <programlisting>
<codeline lineno="1"><highlight class="preprocessor">#ifndef<sp/>_SPECIAL_CODE_H</highlight><highlight class="normal"></highlight></codeline>
<codeline lineno="2"><highlight class="normal"></highlight><highlight class="preprocessor">#define<sp/>_SPECIAL_CODE_H</highlight><highlight class="normal"></highlight></codeline>
<codeline lineno="3"><highlight class="normal"></highlight></codeline>
<codeline lineno="4"><highlight class="normal"></highlight><highlight class="preprocessor">#define<sp/>SPECIAL_CODE(name,<sp/>value,<sp/>doc)<sp/>name<sp/>=<sp/>(value),</highlight><highlight class="normal"></highlight></codeline>
<codeline lineno="5"><highlight class="normal"></highlight></codeline>
<codeline lineno="6"><highlight class="normal"></highlight><highlight class="keyword">enum</highlight><highlight class="normal"><sp/>special_codes</highlight></codeline>
<codeline lineno="7"><highlight class="normal">{</highlight></codeline>
<codeline lineno="8"><highlight class="normal"></highlight><highlight class="preprocessor">#include<sp/>&quot;specialcodes.h&quot;</highlight><highlight class="normal"></highlight></codeline>
<codeline lineno="9"><highlight class="normal">};</highlight></codeline>
<codeline lineno="10"><highlight class="normal"></highlight></codeline>
<codeline lineno="11"><highlight class="normal"></highlight><highlight class="preprocessor">#endif</highlight></codeline>
    </programlisting>
    <location file="specialcode.h"/>
  </compounddef>
</doxygen>
```

**To Reproduce**
Here's the config I used, along with the code under 'Describe the Bug':

```
# doxygen config

GENERATE_XML = YES
GENERATE_HTML = NO
GENERATE_LATEX = NO

ENABLE_PREPROCESSING = YES
MACRO_EXPANSION = YES

INPUT = specialcode.h
```
**Version**
This regressed around 1.8.16. Compiled doxygen from source using latest commit, and bisected to commit 047df30104f628bdfbbe71fbc8d11440a5b0494a

**Comment by albert-github** [comment]
Thanks for the well researched problem and the bisecting!

When looking at the preprocessor output of 1.8.15 (`doxygen -d preprocessor`) we see:
```
Preprocessor output (size: 266 bytes):
---------
00001 // specialcode.h:
00002
00003 #define SPECIAL_CODE(name, value, doc)
00004
00005 enum special_codes
00006 {
00007 # 1 "D:/speeltuin/issue_10273/specialcodes.h" 1
00008 // specialcodes.h:
00009
00010  SC_ORDINARY  = ( 0 ),
00011  SC_SPECIAL  = ( 1 ),
00012  SC_UNKNOWN  = ( 2 ),
00013 # 7 "D:/speeltuin/issue_10273/specialcode.h" 2
00014 };
```
with the version 1.8.16 we see:
```
Preprocessor output (size: 131 bytes):
---------
00001 // specialcode.h:
00002
00003 #define SPECIAL_CODE(name, value, doc)
00004
00005 enum special_codes
00006 {
00007 # 7 "D:/speeltuin/issue_10273/specialcode.h" 2
00008 };
```
One small point of criticism, it would be easier when you would attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
(So it is easier to reproduce and no cut / past problems occur).

**Comment by oumuamua97** [comment]
Thanks Albert, I didn't know about -d preprocessor. Here's a tarball with the files I used (attached[specialcodes.tar.gz](https://github.com/doxygen/doxygen/files/12491842/specialcodes.tar.gz)

**Comment by oumuamua97** [comment]
Hi Albert, could you make an estimate of when this might be addressed?

**Comment by albert-github** [comment]
No idea yet.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
